reconcile/jira_permissions_validator.py

```diff
@@ -256,13 +264,59 @@ def get_jira_boards(query_func: Callable) -> list[JiraBoardV1]:
     ]
 
 
-def run(dry_run: bool, exit_on_permission_errors: bool) -> None:
+def export_boards(boards: list[JiraBoardV1]) -> list[dict]:
+    return [board.dict() for board in boards]
+
+
+def run(
+    dry_run: bool,
+    exit_on_permission_errors: bool,
+    enable_extended_early_exit: bool = False,
+    extended_early_exit_cache_ttl_seconds: int = 3600,
+    log_cached_log_output: bool = False,
+) -> None:
+    gql_api = gql.get_api()
+    boards = get_jira_boards(query_func=gql_api.query)
+    runner_params: RunnerParams = dict(
+        exit_on_permission_errors=exit_on_permission_errors,
+        boards=boards,
+    )
+    if enable_extended_early_exit and get_feature_toggle_state(
+        "jira-permissions-validator-extended-early-exit",
+        default=True,
+    ):
+        vault_settings = get_app_interface_vault_settings()
+        secret_reader = create_secret_reader(use_vault=vault_settings.vault)
+
+        cache_source = CacheSource(
+            boards=export_boards(boards),
+        )
+        extended_early_exit_run(
+            integration=QONTRACT_INTEGRATION,
+            integration_version=QONTRACT_INTEGRATION_VERSION,
+            # don't use `dry_run` in the cache key because this is a read-only integration
+            dry_run=False,
```

Comment:
This means prod run and pr check run will share the same cache, no dry run cache pattern is `/<integration>/<integration_version>/no-dry-run(/<shard>)/latest`, is it ok to let pr check run set cache and cause prod run early exit?

Reply:
That was exactly the idea. I want dry-runs and non-dry-runs using the same cached state

Reply:
I never tested this use case, let's see if this can cause any issues after deploy